include/envoy/stats/histogram.h

```diff
@@ -8,6 +8,10 @@
 #include "envoy/stats/refcount_ptr.h"
 #include "envoy/stats/stats.h"
 
+extern "C" {
+  typedef struct histogram histogram_t;
+}
```

Comment:
I don't think we should leak `histogram_t` into Envoy's API. I think it would be reasonable to add a serialization method to the HistogramStatistics type and use that in the CirconusStatsView.

Reply:
You mean just expose `std::vector`` directly from there?  Makes sense to me

Reply:
I was suggesting that you could add a `serialize` function to HistogramStatistics. That's not too dissimilar from what you've done.

I'm still concerned that this serialization format is tightly coupled to the circonus library.

Reply:
> I was suggesting that you could add a serialize function to HistogramStatistics. That's not too dissimilar from what you've done.

`serialize` on `HistogramStatistics` in full would not meet our needs.  It would capture more than the histogram, I would need to de-serialize just in order to get the `histogram_t` directly.  So one way or the other, it's exposed, it's just the number of layers around it.

> I'm still concerned that this serialization format is tightly coupled to the circonus library

Why?  The interface itself has only ``std::vector`` and nothing else.  Adding an ID wouldn't add anything beyond checking for the concrete class

Reply:
I think it's fine to have the separate methods. I worry that exposing this specific binary format will be a liability if we ever moved to a different library or if the library itself changes its output.